packages/qwik/src/server/prefetch-utils.ts

```diff
@@ -23,9 +23,21 @@ export function prefetchUrlsEventScript(prefetchResources: PrefetchResource[]) {
   const data: QPrefetchData = {
     bundles: flattenPrefetchResources(prefetchResources).map((u) => u.split('/').pop()!),
   };
-  return `document.dispatchEvent(new CustomEvent("qprefetch",{detail:${JSON.stringify(data)}}))`;
+  return `(${PREFETCH_BUNDLES_CODE})(
+    document.currentScript.closest('[q\\\\:container]'),
```

Comment:
Why we need this? `[q\\:container]` is not enough?

Reply:
we need to run this in the browser to use the right one. I used the same logic we have [here](https://github.com/QwikDev/qwik/blob/646165fd5916f54839db2623bd5978efdf822a7b/packages/qwik/src/core/components/prefetch.ts#L81)